src/hotspot/share/opto/vectornode.cpp

```diff
@@ -629,17 +607,7 @@ bool VectorNode::is_vshift_cnt(Node* n) {
 // Check if input is loop invariant vector.
 bool VectorNode::is_invariant_vector(Node* n) {
   // Only Replicate vector nodes are loop invariant for now.
```

Comment:
Random discovery: does this not sound fishy? Can Replicate nodes never be used in a loop variant way? For example with the VectorAPI?

Reply:
Replicate can be variant in Vector API. E.g., we can create an IntVector in loop by IntVector.broadcast (https://github.com/openjdk/jdk/blob/master/src/jdk.incubator.vector/share/classes/jdk/incubator/vector/IntVector.java). We also have such matching rules supporting register as Replicate input.

I only find the use point of this method in Superword, I'm not sure if auto-vectorization has any special handle in vectorizing Replicate.

Reply:
I think this method has no value now. I would remove it and use `n->Opcode() == Op_Replicate` at the two use cases:
`VectorNode::degenerate_vector_rotate` and `SuperWord::output`
That would be more understandable.

In SuperWord, we so far only add Replicate nodes if we have a vector-op that has one arg-slot as all the same input (eg `some_invariant_value` below).
```
for (int i = ...) {
  a[i] = b[i] * some_invariant_value;
}
```
So I guess the assumption is that the Replicate node can always float out of the loop, and hence it would be good to have it on the "right side" (second slot) of the Add/Mul node. This supposedly reduces register pressure. If that is still true I do not know. Hence, I would do this:

```
        if (in1->Opcode() == Op_Replicate && (node_isa_reduction == false) && (n->is_Add() || n->is_Mul())) {
          // We have a Replicate node on in1. Since the input to the Replicate node is expected to be outside
          // the loop, also the Replicate node can float out of the loop. To reduce register pressure, we swap
          // the invariant inpus to the second slot.
          Node* tmp = in1;
          in1 = in2;
          in2 = tmp;
        }
```
Additionally, it would be interesting to add an assert to see if the input of Replicate is indeed loop invariant.

`VectorNode::degenerate_vector_rotate` talks about doing something to replicate nodes anyway, so talking about "invariant" is not helpful.

Reply:
@e1iu what do you think?

Reply:
Agree! I removed this in the latest commit. I'm running full JTReg for it.